openapi-cli/src/test/java/io/ballerina/openapi/CodeGeneratorTest.java

```diff
@@ -475,6 +476,70 @@ public void testCodeGenerationWithoutSchemas() {
         }
     }
 
+    @Test(description = "Test code generation when no schemas given in service")
+    public void testCodeGenerationWithoutSchemasService() {
+        final String serviceName = "no_schema";
+        String definitionPath = RES_DIR.resolve("no_schema.yaml").toString();
+        CodeGenerator generator = new CodeGenerator();
+        try {
+            generator.generateService(definitionPath, serviceName, resourcePath.toString(),  filter, false);
+            if (Files.exists(resourcePath.resolve("no_schema_service.bal")) &&
+                    Files.notExists(resourcePath.resolve("types.bal"))) {
+                Assert.assertTrue(true);
+            } else {
+                Assert.fail("Empty types.bal file has been generated");
+            }
+        } catch (IOException | BallerinaOpenApiException | FormatterException e) {
+            Assert.fail("Error while generating the service. " + e.getMessage());
+        } finally {
+            deleteGeneratedFiles("no_schema_service.bal");
+        }
+    }
+
+    @Test(description = "Test code generation for no schema in given contract but user created type available in " +
+            "service")
+    public void testForGeneratingTypeFileWhenNoSchema() {
+        final String serviceName = "no_schema";
+        String definitionPath = RES_DIR.resolve("no_schema_with_type_bal.yaml").toString();
+        CodeGenerator generator = new CodeGenerator();
+        try {
+            generator.generateService(definitionPath, serviceName, resourcePath.toString(),  filter, false);
+            if (Files.exists(resourcePath.resolve("no_schema_service.bal")) &&
+                    Files.exists(resourcePath.resolve("types.bal"))) {
+                Assert.assertTrue(true);
+            } else {
+                Assert.fail("types.bal file has not been generated");
+            }
+        } catch (IOException | BallerinaOpenApiException | FormatterException e) {
+            Assert.fail("Error while generating the service. " + e.getMessage());
+        } finally {
+            deleteGeneratedFiles("no_schema_service.bal");
+        }
+    }
+
+    @Test(description = "Test code generation when no schemas given in both files (client and service)")
+    public void testCodeGenerationWithoutSchemasBothFiles() {
+        final String serviceName = "no_schema";
+        String definitionPath = RES_DIR.resolve("no_schema.yaml").toString();
+        CodeGenerator generator = new CodeGenerator();
+        try {
+            generator.generateBothFiles(
+                    GeneratorConstants.GenType.GEN_BOTH, definitionPath, serviceName, resourcePath.toString(),  filter
```

Comment:
This is not a good API design IMO. I doubt whether we still need a separate enum argument to indicate what to generate, since you have already defined the method `generateBothFiles` just for that purpose. WDYT?

Reply:
+1 Yeah, here we can remove the enum type since the method clearly indicates the purpose.